File: editorial_ch/chap2-5/prob2-5-4.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		// 輸入
		int N = sc.nextInt();
		
		// 輸出
		for (int i = 2; i <= N; i++) {
			if (isPrime(i) == true) {
				if (i >= 3) System.out.print(" ");
				System.out.print(i);
			}
		}
		System.out.println();
	}
	static boolean isPrime(long N) {
		// 對 2 以上的整數，若 N 為質數時回傳 true，非質數時回傳 false 的函數
		for (long i = 2; i <= N - 1; i++) {
			if (N % i == 0) {
				return false;
			}
		}
		return true;
	}
}

File: editorial_ch/chap2-5/prob2-5-4.cpp
#include <iostream>
using namespace std;

bool isprime(int x) {
	for (int i = 2; i <= x - 1; i++) {
		// 將 x 除以 i 的餘數為 0 時、x 可以被 i 整除
		if (x % i == 0) return false;
	}
	return true;
}

int main() {
	// 輸入
	int N;
	cin >> N;
	
	// 輸出答案（以空白來區隔輸出）
	for (int i = 2; i <= N; i++) {
		if (isprime(i) == true) {
			if (i >= 3) cout << " ";
			cout << i;
		}
	}
	cout << endl;
	return 0;
}